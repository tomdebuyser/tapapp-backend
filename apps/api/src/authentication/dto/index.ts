export * from './authenticated-user.dto';
export * from './login.dto';
