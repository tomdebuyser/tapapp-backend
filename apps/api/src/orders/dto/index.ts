export * from './create-order.dto';
export * from './get-order.dto';
export * from './order-item.dto';
