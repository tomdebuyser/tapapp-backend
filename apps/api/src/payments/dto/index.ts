export * from './create-payment.dto';
export * from './payment-id.dto';
export * from './payconiq-payment.dto';
