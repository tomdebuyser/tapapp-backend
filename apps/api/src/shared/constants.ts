export enum SortDirection {
    Ascending = 'ASC',
    Descending = 'DESC',
}

export type UserSession = {
    userId: string;
    email: string;
    isActive: boolean;
    name: string;
};
