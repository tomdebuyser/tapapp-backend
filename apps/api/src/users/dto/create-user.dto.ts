import { IsEmail, IsOptional, ArrayNotEmpty, IsUUID } from 'class-validator';

import { IsName } from '@libs/common/validators';

export class CreateUserRequest {
    @IsEmail()
    readonly email: string;

    @IsUUID('4', { each: true })
    @ArrayNotEmpty()
    readonly roleIds: string[];

    @IsOptional()
    @IsName()
    readonly firstName?: string;

    @IsOptional()
    @IsName()
    readonly lastName?: string;
}
