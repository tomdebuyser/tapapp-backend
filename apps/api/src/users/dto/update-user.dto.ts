import { IsUUID, IsOptional } from 'class-validator';
import { ApiProperty } from '@nestjs/swagger';

import { IsName } from '@libs/common/validators';

export class UpdateUserRequest {
    @IsOptional()
    @IsName()
    readonly firstName?: string;

    @IsOptional()
    @IsName()
    readonly lastName?: string;
}

export class UserIdParam {
    @ApiProperty({ required: true })
    @IsUUID('4')
    readonly userId: string;
}
