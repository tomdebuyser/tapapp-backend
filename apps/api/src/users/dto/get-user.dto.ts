import { UserState } from '@libs/models';
import { AuditedEntityResponse } from '../../shared/dto';

export class UserResponse extends AuditedEntityResponse {
    readonly email: string;
    readonly state: UserState;
    readonly roles: UserResponseRole[];
    readonly firstName?: string;
    readonly lastName?: string;
}

class UserResponseRole {
    readonly id: string;
    readonly name: string;
}
