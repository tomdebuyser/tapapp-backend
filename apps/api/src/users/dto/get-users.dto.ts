import { IsOptional, IsEnum } from 'class-validator';

import { PagingQuery, PagingMeta } from '../../shared/dto';
import { UserResponse } from './get-user.dto';

export enum UsersSortColumns {
    Email = 'email',
    IsActive = 'isActive',
    CreatedAt = 'createdAt',
    UpdatedAt = 'updatedAt',
    CreatedBy = 'createdBy',
    UpdatedBy = 'updatedBy',
    Name = 'name',
}

export class GetUsersRequestQuery extends PagingQuery {
    @IsOptional()
    @IsEnum(UsersSortColumns)
    readonly sortBy?: UsersSortColumns;
}

export class GetUsersResponse {
    readonly meta: PagingMeta;
    readonly data: UserResponse[];
}
