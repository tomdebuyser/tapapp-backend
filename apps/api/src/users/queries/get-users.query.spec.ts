import { Test, TestingModule } from '@nestjs/testing';

import { ModelsModule, UserRepository } from '@libs/models';
import { GetUsersRequestQuery, UsersSortColumns } from '../dto';
import { SortDirection } from '../../shared/constants';
import { Config } from '../../config';
import { GetUsersHandler } from './get-users.query';

describe('GetUsersHandler', () => {
    let module: TestingModule;
    let userRepository: UserRepository;
    let handler: GetUsersHandler;

    beforeAll(async () => {
        module = await Test.createTestingModule({
            imports: [ModelsModule.registerTest(Config.models)],
            providers: [GetUsersHandler],
        }).compile();

        handler = module.get(GetUsersHandler);
        userRepository = module.get(UserRepository);
    });

    afterAll(async () => {
        await userRepository.manager.connection.close();
        await module.close();
    });

    describe('execute', () => {
        const testQueries: GetUsersRequestQuery[] = [
            {},
            { skip: 2, take: 2 },
            { search: 'wltc1' },
            { search: 'WLTC2' },
            { search: 'Tom' },
            {
                sortBy: UsersSortColumns.CreatedAt,
                sortDirection: SortDirection.Ascending,
            },
        ];

        testQueries.forEach(request => {
            it(`should return a paged list of users: ${JSON.stringify(
                request,
            )}`, async () => {
                const result = await handler.execute({ data: request });
                expect(result).toMatchSnapshot();
            });
        });

        it('should return an empty list if no users found', async () => {
            const result = await handler.execute({
                data: { search: 'nonsensicaljibberish' },
            });
            expect(result).toMatchSnapshot();
        });
    });
});
