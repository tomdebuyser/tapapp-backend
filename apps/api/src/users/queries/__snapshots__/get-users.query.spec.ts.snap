// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetUsersHandler execute should return a paged list of users: {"search":"Tom"} 1`] = `
Object {
  "data": Array [
    User {
      "createdAt": 2020-03-03T09:30:14.640Z,
      "createdBy": null,
      "email": "tom.debuyser@gmail.com",
      "id": "e810ee33-6cd3-41dd-8755-6abba3bb9efa",
      "isActive": true,
      "name": "Tom De Buyser",
      "updatedAt": 2020-03-03T09:30:18.095Z,
      "updatedBy": null,
    },
  ],
  "meta": Object {
    "count": 1,
    "skip": 0,
    "totalCount": 1,
  },
}
`;

exports[`GetUsersHandler execute should return a paged list of users: {"search":"WLTC2"} 1`] = `
Object {
  "data": Array [
    User {
      "createdAt": 2020-03-03T09:30:32.473Z,
      "createdBy": "tom.debuyser@gmail.com",
      "email": "wltc2@tapapp.be",
      "id": "45c2c554-b06a-4209-aba8-b097cb5f8566",
      "isActive": true,
      "name": "User 2",
      "updatedAt": 2020-03-03T09:30:36.284Z,
      "updatedBy": "tom.debuyser@gmail.com",
    },
  ],
  "meta": Object {
    "count": 1,
    "skip": 0,
    "totalCount": 1,
  },
}
`;

exports[`GetUsersHandler execute should return a paged list of users: {"search":"wltc1"} 1`] = `
Object {
  "data": Array [
    User {
      "createdAt": 2020-03-03T09:30:24.076Z,
      "createdBy": "tom.debuyser@gmail.com",
      "email": "wltc1@tapapp.be",
      "id": "49affd08-f461-4e80-899f-7612e971ff12",
      "isActive": true,
      "name": "User 1",
      "updatedAt": 2020-03-03T09:30:27.518Z,
      "updatedBy": "tom.debuyser@gmail.com",
    },
  ],
  "meta": Object {
    "count": 1,
    "skip": 0,
    "totalCount": 1,
  },
}
`;

exports[`GetUsersHandler execute should return a paged list of users: {"skip":2,"take":2} 1`] = `
Object {
  "data": Array [
    User {
      "createdAt": 2020-03-03T09:30:41.190Z,
      "createdBy": "tom.debuyser@gmail.com",
      "email": "wltc3@tapapp.be",
      "id": "db5e788c-2642-4c40-a63a-f13675048ddc",
      "isActive": true,
      "name": "User 3",
      "updatedAt": 2020-03-03T09:30:44.265Z,
      "updatedBy": "tom.debuyser@gmail.com",
    },
    User {
      "createdAt": 2020-03-03T09:30:32.473Z,
      "createdBy": "tom.debuyser@gmail.com",
      "email": "wltc2@tapapp.be",
      "id": "45c2c554-b06a-4209-aba8-b097cb5f8566",
      "isActive": true,
      "name": "User 2",
      "updatedAt": 2020-03-03T09:30:36.284Z,
      "updatedBy": "tom.debuyser@gmail.com",
    },
  ],
  "meta": Object {
    "count": 2,
    "skip": 2,
    "totalCount": 6,
  },
}
`;

exports[`GetUsersHandler execute should return a paged list of users: {"sortBy":"createdAt","sortDirection":"ASC"} 1`] = `
Object {
  "data": Array [
    User {
      "createdAt": 2020-03-03T09:30:14.640Z,
      "createdBy": null,
      "email": "tom.debuyser@gmail.com",
      "id": "e810ee33-6cd3-41dd-8755-6abba3bb9efa",
      "isActive": true,
      "name": "Tom De Buyser",
      "updatedAt": 2020-03-03T09:30:18.095Z,
      "updatedBy": null,
    },
    User {
      "createdAt": 2020-03-03T09:30:24.076Z,
      "createdBy": "tom.debuyser@gmail.com",
      "email": "wltc1@tapapp.be",
      "id": "49affd08-f461-4e80-899f-7612e971ff12",
      "isActive": true,
      "name": "User 1",
      "updatedAt": 2020-03-03T09:30:27.518Z,
      "updatedBy": "tom.debuyser@gmail.com",
    },
    User {
      "createdAt": 2020-03-03T09:30:32.473Z,
      "createdBy": "tom.debuyser@gmail.com",
      "email": "wltc2@tapapp.be",
      "id": "45c2c554-b06a-4209-aba8-b097cb5f8566",
      "isActive": true,
      "name": "User 2",
      "updatedAt": 2020-03-03T09:30:36.284Z,
      "updatedBy": "tom.debuyser@gmail.com",
    },
    User {
      "createdAt": 2020-03-03T09:30:41.190Z,
      "createdBy": "tom.debuyser@gmail.com",
      "email": "wltc3@tapapp.be",
      "id": "db5e788c-2642-4c40-a63a-f13675048ddc",
      "isActive": true,
      "name": "User 3",
      "updatedAt": 2020-03-03T09:30:44.265Z,
      "updatedBy": "tom.debuyser@gmail.com",
    },
    User {
      "createdAt": 2020-03-03T09:30:49.457Z,
      "createdBy": "tom.debuyser@gmail.com",
      "email": "wltc4@tapapp.be",
      "id": "c356fce8-9f95-4c69-9e71-8f4d74f49b8f",
      "isActive": true,
      "name": "User 4",
      "updatedAt": 2020-03-03T09:30:52.420Z,
      "updatedBy": "tom.debuyser@gmail.com",
    },
    User {
      "createdAt": 2020-03-03T09:30:56.536Z,
      "createdBy": "tom.debuyser@gmail.com",
      "email": "wltc5@tapapp.be",
      "id": "af965f04-8c7d-4e20-b230-932b0ff3a33f",
      "isActive": false,
      "name": "User 5",
      "updatedAt": 2020-03-03T09:31:00.109Z,
      "updatedBy": "tom.debuyser@gmail.com",
    },
  ],
  "meta": Object {
    "count": 6,
    "skip": 0,
    "totalCount": 6,
  },
}
`;

exports[`GetUsersHandler execute should return a paged list of users: {} 1`] = `
Object {
  "data": Array [
    User {
      "createdAt": 2020-03-03T09:30:56.536Z,
      "createdBy": "tom.debuyser@gmail.com",
      "email": "wltc5@tapapp.be",
      "id": "af965f04-8c7d-4e20-b230-932b0ff3a33f",
      "isActive": false,
      "name": "User 5",
      "updatedAt": 2020-03-03T09:31:00.109Z,
      "updatedBy": "tom.debuyser@gmail.com",
    },
    User {
      "createdAt": 2020-03-03T09:30:49.457Z,
      "createdBy": "tom.debuyser@gmail.com",
      "email": "wltc4@tapapp.be",
      "id": "c356fce8-9f95-4c69-9e71-8f4d74f49b8f",
      "isActive": true,
      "name": "User 4",
      "updatedAt": 2020-03-03T09:30:52.420Z,
      "updatedBy": "tom.debuyser@gmail.com",
    },
    User {
      "createdAt": 2020-03-03T09:30:41.190Z,
      "createdBy": "tom.debuyser@gmail.com",
      "email": "wltc3@tapapp.be",
      "id": "db5e788c-2642-4c40-a63a-f13675048ddc",
      "isActive": true,
      "name": "User 3",
      "updatedAt": 2020-03-03T09:30:44.265Z,
      "updatedBy": "tom.debuyser@gmail.com",
    },
    User {
      "createdAt": 2020-03-03T09:30:32.473Z,
      "createdBy": "tom.debuyser@gmail.com",
      "email": "wltc2@tapapp.be",
      "id": "45c2c554-b06a-4209-aba8-b097cb5f8566",
      "isActive": true,
      "name": "User 2",
      "updatedAt": 2020-03-03T09:30:36.284Z,
      "updatedBy": "tom.debuyser@gmail.com",
    },
    User {
      "createdAt": 2020-03-03T09:30:24.076Z,
      "createdBy": "tom.debuyser@gmail.com",
      "email": "wltc1@tapapp.be",
      "id": "49affd08-f461-4e80-899f-7612e971ff12",
      "isActive": true,
      "name": "User 1",
      "updatedAt": 2020-03-03T09:30:27.518Z,
      "updatedBy": "tom.debuyser@gmail.com",
    },
    User {
      "createdAt": 2020-03-03T09:30:14.640Z,
      "createdBy": null,
      "email": "tom.debuyser@gmail.com",
      "id": "e810ee33-6cd3-41dd-8755-6abba3bb9efa",
      "isActive": true,
      "name": "Tom De Buyser",
      "updatedAt": 2020-03-03T09:30:18.095Z,
      "updatedBy": null,
    },
  ],
  "meta": Object {
    "count": 6,
    "skip": 0,
    "totalCount": 6,
  },
}
`;

exports[`GetUsersHandler execute should return an empty list if no users found 1`] = `
Object {
  "data": Array [],
  "meta": Object {
    "count": 0,
    "skip": 0,
    "totalCount": 0,
  },
}
`;
